### ISSUE TITLE
Privacy Issue: Why is nMobile.apk using 4 trackers that are sending user data to 3rd parties?

### ISSUE THREAD

**Issue by mvozzo** [nknorg/nMobile#137]
Hello NKN.org developers,  First of all, thank you for offering a .apk for download. 

In this world of increased surveillance, privacy, security and decentralisation (3 of the main pillars of your app and of NKN) are becoming increasingly important. I have made the switch to a de-Googled phone and hence really appreciate the .apk file. 

You're app prides itself on being a secure, private and encrypted messaging platform, yet when I run the .APK file I've downloaded from your website through a tool called ClassyShark3xodus ( https://f-droid.org/en/packages/com.oF2pks.classyshark3xodus/ ), the following 4 x 3rd party trackers were detected in the .apk:

1. CleverTap  - 16com.clevertap  (not sure what data being collected) 
2. Google AdMob  - Google's Mobile Ad Platform
3. Google Firebase Analytics  - Google Analytics for Mobile
4. 2?Sentry  -  441io.sentry  (not sure what data being collected)   

I have the following 3 questions:
Can you please outline why these trackers are needed in an app that is meant to protect users privacy?

Do you realise that by including these 3rd party trackers you are sending data to centralised entities?  Doesn't this go against one of your core value propositions of being decentralised? 

When can see a version of the nMobile.APK with all 3rd party trackers removed?

Thanks and keep up the great work. I will be fully supportive of this project once these issues have been addressed.

**Comment by 002gRTEah3mRYH** [comment]
I also mentionned some other problems in another issue, but yeah, feels like strange

**Comment by jiangzhiguo1992** [comment]
Currently there are indeed 4 trackers in the project.

1.CleverTap - used to send APNS push notifications for ios terminals.
2. Google AdMob - Currently we don't know which library introduced this tracker, we will remove it in subsequent versions.
3.Google Firebase - FCM notification push for sending android terminals.
4. Sentry - Used to collect terminal crash logs, we are still considering whether to remove it.

Thank you very much for your feedback.

**Comment by gabriellemon** [comment]
I use nMobile on IOS and I don't get push notifications from the app. Does anyone else? Also, have checked all permissions and such, should be receiving it.

**Comment by gabriellemon** [comment]
> 

I think Sentry is fine since it's open source. If a f-droid version is made, it should be removed though. Ios and play store makes sense due to the varying devices and OS support. Wouldn't want bad reviews due to bugs we couldn't track. The f-droid community would come here to GitHub and help with resolving issues.

**Comment by jiangzhiguo1992** [comment]
Regarding nMobile's message notification push, only after sending your own deviceToken to the other party can you receive the push from the other party, which is operated in the interface.

**Comment by jiangzhiguo1992** [comment]
We haven't made an f-droid version yet. However, we are considering whether to release a version without trackers at all.

**Comment by johdough** [comment]
Great sting on a really ideal app! THANK ya'll!
One I LIKE IT! The concept is GREAT!
NOW....my vote to remove all the xTra trackers... like on my un-googirled phone!
And...
1. where is any INFO on "setting up an account," since it asks for... ENTER WALLET NAME....do we need a wallet name already, or could we make one UP??
And can we do so without KYC?

I have not located any HOW TOs... for just getting it "account-able" setup???

Did a ten minuted search on 3 search engines (private only!) and they whey are so private I didn't find any Step by stepper, situation!  LOL!
So... please where's the "HOW TO..... info!

Thanks again for your app.....

**Comment by jiangzhiguo1992** [comment]
> Great sting on a really ideal app! THANK ya'll! One I LIKE IT! The concept is GREAT! NOW....my vote to remove all the xTra trackers... like on my un-googirled phone! And...
> 
> 1. where is any INFO on "setting up an account," since it asks for... ENTER WALLET NAME....do we need a wallet name already, or could we make one UP??
>    And can we do so without KYC?
> 
> I have not located any HOW TOs... for just getting it "account-able" setup???
> 
> Did a ten minuted search on 3 search engines (private only!) and they whey are so private I didn't find any Step by stepper, situation! LOL! So... please where's the "HOW TO..... info!
> 
> Thanks again for your app.....

The setting of the wallet name can be customized when the wallet is created. Or on the details page of the wallet, click on the wallet name to modify it. The wallet name is more like a note, you can modify it as you like.
